### ISSUE TITLE
peculiar outfld behavior when passed array of length ncol

### ISSUE THREAD

**Issue by worleyph** [E3SM-Project/E3SM#2549]
This is a follow-on to issue #2548 . That issue was resolved by setting all columns in a chunk to (the same) ieflx_glob value instead of just the "active" columns, and then calling outfld with the full array and specifying the dimension as pcols.

An alternate approach that we tried, and which failed, was

     real(r8) :: ieflx(pcols)

     do lchnk = begchunk, endchunk
         ieflx(:)   = -1._r8
         ieflx(1:ncol) = ieflx_glob
         call outfld('IEFLX', ieflx(1:ncol), ncol, lchnk)
     enddo

Even though only the 1:ncol array segment is used in outfld and the dimension is declared to be ncol, '-1' values still show up in the netcdf file. (If initialized to 0, then 0's show up). Moreover, the locations of the '-1' values (or '0' values) change with the PE layout. This is true for hash f0c440b22  (so before the PR that changed how chumks are assigned) as well as for current master.

While the majority of the calls to outfld use 'pcols' as the dimension, there are some that use 'ncol'. I also do not know how the nonactive columns are set when passing in the whole array and specifying pcols. Seems like the above issue would affect a lot more than IEFLX, so I really have no idea what is going. However, the above seems to indicate that something is not working the way that it should in outfld. It would be worthwhile if an outfld expert could speak to this.

TL/DR summary: outfld is using array elements that it shouldn't, and placing them in the netcdf output file in a PE-layout sensitive way.

**Comment by worleyph** [comment]
(Note edit is previous comment: comparing

    <value compclass="ATM">32</value>

to

    <value compclass="ATM">33</value>

though just changing the load balancing option might demonstrate this as well. I have not checked that yet.

**Comment by worleyph** [comment]
Can also reproduce by changing phys_loadbalance from 2 (the default) to 0, without having to change the PE layout.
